Framework/Kernel/test/TimeSeriesPropertyTest.h

```diff
@@ -548,6 +548,87 @@ class TimeSeriesPropertyTest : public CxxTest::TestSuite {
     delete log;
   }
 
+  //----------------------------------------------------------------------------
+  void test_makeFilterByValueWithROI() {
+    TimeSeriesProperty<double> *log = new TimeSeriesProperty<double>("MyIntLog");
+    TS_ASSERT_THROWS_NOTHING(log->addValue("2007-11-30T16:17:00", 1));
+    TS_ASSERT_THROWS_NOTHING(log->addValue("2007-11-30T16:17:10", 2));
+    TS_ASSERT_THROWS_NOTHING(log->addValue("2007-11-30T16:17:20", 3));
+    TS_ASSERT_THROWS_NOTHING(log->addValue("2007-11-30T16:17:30", 2.0));
+    TS_ASSERT_THROWS_NOTHING(log->addValue("2007-11-30T16:17:40", 2.01));
+    TS_ASSERT_THROWS_NOTHING(log->addValue("2007-11-30T16:17:50", 6));
+
+    TS_ASSERT_EQUALS(log->realSize(), 6);
+    TimeInterval expandedTime(DateAndTime(0), DateAndTime(1));
+
+    // Test centred log value boundaries
+    TimeROI roi = log->makeFilterByValue(1.8, 2.2, false, expandedTime, 1.0, true);
+
+    TS_ASSERT_EQUALS(roi.numBoundaries(), 4);
+    DateAndTime expected;
+
+    expected = DateAndTime("2007-11-30T16:17:09");
```

Comment:
Why you are setting expected as a temporary variable rather than just hard coding in the test compare? I don't see what the advantage is